keps/sig-cluster-lifecycle/kubeadm/20190424-kubeadm-for-windows.md

```diff
@@ -115,23 +115,35 @@ The motivation for this KEP is to provide a tool to allow users to take a Window
+1. The user will download a set of required binaries (kubelet, kubeadm, and wins) using a script.
 
-2. The script will register kubelet and kube-proxy as Windows services.
+1. The script will register kubelet as a Windows service.
 
-3. The script will upload a default ConfigMap to the cluster for the default Windows KubeletConfiguration if one does not already exist.
+1. The user will run "kubeadm join ..." to join the node to the cluster. In this step kubeadm will run preflight checks and proceed with the regular join process.
 
-4. The script will run "kubeadm join ..." to join the node to the cluster. In this step kubeadm will run preflight checks and proceed with the regular join process.
+1. kubeadm will restart the kubelet service using flags that kubeadm fed to the Windows service. kubeadm will proceed to bootstrap the node. After this process is finished the node should show with the status of NotReady.
 
-5. kubeadm will restart the kubelet service using flags that kubeadm fed to the Windows service. kubeadm will proceed to bootstrap the node. After this process is finished the node should show with the status of NotReady.
+1. The user will then deploy the Flannel and kube-proxy DaemonSets. These will run and initialize container networking.
 
-6. The same script will then configure FlannelD. The script will (re-)register FlannelD as a service and start it with the correct configuration, optionally using parameters from kubeadm.
+1. The node status should become Ready.
 
-7. kube-proxy will do the same steps as FlannelD shortly after.
+### Implementation Details/Notes/Constraints
 
-8. The node status should be Ready.
+#### Wins used to enable privileged/host network DaemonSets
 
-### Implementation Details/Notes/Constraints
+Windows has no native support for creating privileged containers or attaching them to the host network.
+[Wins](https://github.com/rancher/wins/) is a project from Rancher that works around this shortcoming by exposing an API
+to run processes on the host. This API can be exposed to containers via a [named pipe](https://docs.microsoft.com/en-us/windows/win32/ipc/named-pipes)
+to allow those containers to launch processes that escape the restrictions normally imposed by Windows containerization.
+Additional details about this approach can be found [here](https://docs.google.com/document/d/1dXLs2XR8tqueSYWxAb0OGzKqKzx1pR6l2b1JXXT8kqA/edit?usp=sharing).
+
+#### Provisioning script creates a fake host network
+
+In order for the Flannel and kube-proxy DaemonSets to run before CNI has been initialized, they need to be running with `hostNetwork: true` on their Pod specs. This is the established pattern on Linux for bootstrapping CNI, and we are utilizing it here as well. This is in spite of the fact that our containers will not actually need networking at all since the actual Flannel/kube-proxy process will be running outside of the container through wins.
```

Comment:
How will kube-proxy access it's service account credentials?

Reply:
this is currently done like so:
https://github.com/benmoss/kubeadm-windows/blob/master/kube-proxy/kube-proxy.yml#L10

the KubeProxyConfiguration has:
`certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt`